libs/npm/core/src/utils/elastic/generateQueryValues.ts

```diff
@@ -28,13 +29,16 @@ export const generateQueryValues = (
     generateTerms('sourceId', settings.sourceIds),
     generateTerms('mediaTypeId', settings.mediaTypeIds),
     generateTerms('seriesId', settings.seriesIds),
-    generateTerms('contributorId', settings.contributorIds),
+    // Following line replaced by generateShouldQuery:
+    // generateTerms('contributorId', settings.contributorIds),
```

Comment:
Do we actually want to remove this?  When searching for `contributorId` I would think this would still be the right thing.

Reply:
If the contributorId is not in content.contributor_id, we would not get any result. For instance, if we have John Doe as byline but it is not included in content.contributor_id, we would not get the record, it is handled as `AND` statement. Should we handle as 'OR' as we are doing with byline?

Reply:
If the user is making a search for content with a single `contributorId` then it should be required.  If they are searching for multiple then it would be a `should` (or).

Reply:
At the moment the line `generateTerms('contributorId', settings.contributorIds),` is generating a list of IDs:
```
{
                    "terms": {
                        "contributorId": [
                            6,
                            8,
                            9
                        ]
                    }
                },
```
I would need to add the logic you mention.